### DISCUSSION TITLE
use file browse to create a file in a very nested directory tree

### DISCUSSION THREAD

**Discussion by gozes** [nvim-telescope/telescope.nvim#1438] [Q&A] [answered]
Does anyone know of a good way to make the file browser search for more than just the root directories? I often use it to create a file but if I need to create a file in a very nested directory I have to "walk" the directories before I can create a file. It will be nice if I could just search for the directory, go in it and then use C-e to create the file.

**Comment by l-kershaw** [comment, marked as answer]
You can adjust the `depth` option for `file_browser`.
See `:h builtin.file_browser()` for more details 🙂

**Comment by fdschmidt93** [comment]
I'm not sure `depth`  ~~helps~~ is optimal here because it becomes fuzzy as to where you want to create the file. Eventually, https://github.com/nvim-telescope/telescope.nvim/pull/1257 (either as part of telescope.nvim or as an extension) will largely extend the file browser and offer a toggleable folder browser pretty much exactly for such a use case.

**Reply by l-kershaw** [reply]
I'm not sure I agree. Using a higher `depth` value, you can:
- search for the folder you want
- open it
- create a file in that folder in the usual way.

This seems to be all that @gozes is asking for, unless I am misunderstanding something?

**Reply by fdschmidt93** [reply]
Sorry, I should have elaborated more. Of course the simple scenario works with higher `depth` but in my opinion it is an incomplete user experience.

I think it is preferred to have `depth=1` to maintain the logical separation between files in a folder (`file_browser`) and folders (`folder_browser`) which lends itself better to typical file system operations. Furthermore, some questions remain - to name the most obvious ones:

* with higher `depth`, AFAIK, you currently cannot easily go upwards in tree once you've entered a directory
* unclear how high `depth` should be

**Reply by gozes** [reply]
well at the moment I have set `depth` to be 5 and testing that out after @l-kershaw  comment. I do agree with you @fdschmidt93 that what you show in #1257 is 80ish % of what I want for my use case. @l-kershaw hit the nail on the head with his understanding of my question. I guess a better user story for this will be something like

**As a** user of telescope file browser
**I want** to be able to search for a directory that maybe be a few levels deep from the root of the project
**So that** I can create a file or a directory and a file under it

**Reply by fdschmidt93** [reply]
>  #1257 is 80ish % of what I want for my use case

I'm curious, what are the other 20%?

**Reply by gozes** [reply]
So in the video at the top of the issue, you searched for a directory, went in it, and then move around to go into another directory. What I would have liked is to go from root -> search for directory -> create the file in that directory. I could be looking at it wrong but that is what I gather from watching the clip a few times. Please do correct me if I'm getting it wrong :)

**Reply by fdschmidt93** [reply]
As #1257 stands, you 

root -> search for directory -> __enter directory__ -> create file -> possibly go else where with `folder_browser`

#1257 also uses `vim.fn.input` rather than parsing the `prompt` when creating a file (for more coherent experience); so what could be done is parsing the selected entry and then suggesting to create a file there when creating a file in the folder browser.

**Reply by gozes** [reply]
ok that workflow definitely work for me. Better then having to enter each directory one at time to get to were I want just to create a file `root -> search for directory -> enter directory -> create file -> possibly go else where with folder_browser`

**Reply by gozes** [reply]
For creating a file the current way with `C-e` works perfectly fine for me. getting to the directory so that I can then  `C-e` is the issue IMO
